x-pack/plugins/infra/public/alerting/metric_threshold/mocks/metric_threshold_rule.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Rule } from '@kbn/alerting-plugin/common';
+import { v4 as uuidv4 } from 'uuid';
+import { Aggregators, Comparator } from '../../../../common/alerting/metrics';
+import { MetricThresholdRuleTypeParams } from '..';
+
+export const buildMetricThresholdRule = (
+  rule: Partial<Rule<MetricThresholdRuleTypeParams>> = {}
+): Rule<MetricThresholdRuleTypeParams> => {
+  return {
+    alertTypeId: 'metrics.alert.threshold',
```

Comment:
@ymao1 Not sure if this [comment](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/common/rule.ts#L140) was from you, but I see the last commit was by you:

```
// this is persisted in the Rule saved object so we would need a migration to change this to ruleTypeId
```

When I logged this information in the browser, I saw `ruleTypeId`, not `alertTypeId`. Does it mean we can change the type? What does migration mean in the comment?

Reply:
Hi @maryam-saeidi. Do you know what the source of the rule was that you logged? Was it from the alerting APIs? We have to keep `alertTypeId` around because that is the name of the field in the saved object. In order to rename that field, we would have to do a migration to change all the rules but we have not done that yet. I know when we get the rule and return it via the alerting API, we rename alertTypeId to `rule_type_id` so maybe there is another transformation that occurs to change `rule_type_id` to `ruleTypeId` whereever you're getting the rule from?

Reply:
Thanks for the info, I will check later to see where the transformation is happening 👍🏻

Reply:
This is the API that we use to fetch rule: https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/rules_client/methods/get.ts#L21

I don't see any renaming other than removing `apiKey`